experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/unboxed.spec.ts

```diff
@@ -211,6 +279,254 @@ describe("unboxed unit tests", () => {
 				assert.equal(unboxed.value, true);
 			});
 		});
+
+		describe("Value field", () => {
+			it("Boolean", () => {
+				const { fieldSchema, context, cursor } = createValueLeafTree(
+					ValueSchema.Boolean,
+					true,
+				);
+				assert.equal(unboxedField(context, fieldSchema, cursor), true);
+			});
+
+			it("Number", () => {
+				const { fieldSchema, context, cursor } = createValueLeafTree(
+					ValueSchema.Number,
+					42,
+				);
+				assert.equal(unboxedField(context, fieldSchema, cursor), 42);
+			});
+
+			it("String", () => {
+				const { fieldSchema, context, cursor } = createValueLeafTree(
+					ValueSchema.String,
+					"Hello world",
+				);
+				assert.equal(unboxedField(context, fieldSchema, cursor), "Hello world");
+			});
+
+			// TODO: Fluid Handle
+
+			it("Struct", () => {
+				const builder = new SchemaBuilder("test");
+				const stringLeafSchema = builder.leaf("string", ValueSchema.String);
+				const booleanLeafSchema = builder.leaf("boolean", ValueSchema.Boolean);
+				const structSchema = builder.struct("struct", {
+					foo: SchemaBuilder.fieldValue(stringLeafSchema),
+					bar: SchemaBuilder.fieldSequence(booleanLeafSchema),
+				});
+				const rootSchema = SchemaBuilder.field(FieldKinds.value, structSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const initialTree = {
+					foo: "Hello world",
+					bar: [true, false, true],
+				};
+
+				const { context, cursor } = initializeTreeWithContent(schema, initialTree);
+
+				const unboxed = unboxedField(context, rootSchema, cursor);
+
+				assert(unboxed !== undefined);
+				assert.equal(unboxed.foo, "Hello world");
+				assert.equal(unboxed.bar.length, 3);
+				assert.equal(unboxed.bar.at(0), true);
+				assert.equal(unboxed.bar.at(1), false);
+				assert.equal(unboxed.bar.at(2), true);
+			});
+
+			it("Recursive struct", () => {
+				const builder = new SchemaBuilder("test");
+				const stringLeafSchema = builder.leaf("string", ValueSchema.String);
+				const structSchema = builder.structRecursive("struct", {
+					name: SchemaBuilder.fieldValue(stringLeafSchema),
+					child: SchemaBuilder.fieldRecursive(FieldKinds.optional, () => structSchema),
+				});
+				const rootSchema = SchemaBuilder.field(FieldKinds.value, structSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const initialTree = {
+					name: "Foo",
+					child: {
+						name: "Bar",
+						child: undefined,
+					},
+				};
+
+				const { context, cursor } = initializeTreeWithContent(schema, initialTree);
+
+				const unboxed = unboxedField(context, rootSchema, cursor);
+
+				assert(unboxed !== undefined);
+				assert.equal(unboxed.name, "Foo");
+				assert(unboxed.child !== undefined);
+				assert.equal(unboxed.child.name, "Bar");
+				assert.equal(unboxed.child.child, undefined);
+			});
+
+			it("Union", () => {
```

Comment:
if you want to test unboxedUnion, it could have its own suite. Grouping in in here is kinda odd, special as just "union" since all the other ones technically still use a union (of one thing).

Reply:
This case has been removed.